snuba/datasets/tags_column_processor.py

```diff
@@ -148,6 +148,12 @@ def __tag_expr(self, parsed_col: ParsedNestedColumn, table_alias: str = "",) ->
             if actual_tag in self.__promoted_columns[col]:
                 return qualified_column(self.__string_col(actual_tag), table_alias)
 
+            actual_tag = self.__get_tag_column_map()[col].get(
+                tag_name.replace(".", "_"), tag_name
+            )
+            if actual_tag in self.__promoted_columns[col]:
+                return qualified_column(self.__string_col(actual_tag), table_alias)
```

Comment:
This is not the right way to fix this behavior. 
Promoted columns having the same name of the original tag/context with dots replaced by underscore is a coincidence of the events and transactions tables. This is not supposed to be the default behavior applied during tag processing.

The right place to do this is in the events dataset itself:
https://github.com/getsentry/snuba/blob/master/snuba/datasets/events.py#L372-L377
Where the tags in the form `tag_name.tag_inner_name` are mapped into `tag_name_tag_inner_name`. 
You should add a similar mapping for contexts.